### PR TITLE
docs: ADR: Specificity in database column references

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-1680/adr-specify-tables-when-querying-columns

This PR includes my suggestion of a "Specificity in database column references" ADR.

The ADR proposes a standard for explicitly specifying the full table name or alias for each column in our SQL queries, especially for queries with joins between multiple tables. This decision is a direct response to recent ambiguity errors encountered due to a database migration, emphasizing the need for clearer SQL query standards in complex joins and schema changes.

### CODE REVIEW

**Review by gardleopard — APPROVED** [review]
Great! LGTM:)

File: website/docs/contributing/ADRs/back-end/specificity-db-columns.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+## Background
+
+We recently experienced an issue where a database migration that introduced a new column resulted in ambiguity errors in our application, which highlighted the need for clearer SQL query standards. Currently, our queries often reference columns without specifying their parent tables, leading to potential ambiguity in complex queries that join multiple tables. This issue becomes more pronounced during database schema changes and migrations, where ambiguity in column references can lead to hard-to-anticipate runtime errors.
+
+## Decision
+
+To mitigate these risks, we will adopt a standard of explicitly specifying the full table name or alias for each column in our SQL queries. This standard is not just about improving readability, but is crucial for avoiding ambiguity in queries, especially when performing joins between tables. The decision to use the full table name or an alias will be left to the discretion of the developer, with a focus on maximizing clarity and maintainability.
+
```
Comment (Tymek) [inline website/docs/contributing/ADRs/back-end/specificity-db-columns.md:12]: Quick example of preferred code and discouraged syntax could make it easier to read, process and understand.

**Review by ivarconr — APPROVED** [review]

**Comment by ivarconr** [comment]
Great writeup @nunogois ! 👍🏼

File: website/docs/contributing/ADRs/back-end/specificity-db-columns.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+The adoption of this practice comes with minimal concerns:
+
+1. **Slight increase in verbosity**: The queries will be slightly longer due to the addition of table names in columns. However, the benefits in clarity and maintainability far outweigh this minor increase in verbosity.
+
+2. **Initial adaptation curve**: There might be a brief period of adjustment as developers adapt to this new standard. However, given its straightforward nature, this learning curve is expected to be minimal.
+
+## Conclusion
+
+The adoption of explicit table name specification in our SQL queries is a strategic decision aimed at improving the clarity, maintainability, and reliability of our database interactions. This practice will ensure that our queries remain robust and clear, especially in the context of evolving database schemas and complex query scenarios. We will gradually implement this standard in our existing codebase by following the Boy Scout Rule. This change aligns with our commitment to writing clean, understandable, and maintainable code, thereby enhancing the overall quality of our software development processes.
```
Comment (kwasniew) [inline website/docs/contributing/ADRs/back-end/specificity-db-columns.md:33]:
```suggestion
The adoption of explicit table name specification in our SQL queries is a strategic decision aimed at improving the clarity, maintainability, and reliability of our database interactions. This practice will ensure that our queries remain robust and clear, especially in the context of evolving database schemas and complex query scenarios. We will gradually implement this standard in our existing codebase by following the Girl Scout Rule. This change aligns with our commitment to writing clean, understandable, and maintainable code, thereby enhancing the overall quality of our software development processes.
```

File: website/docs/contributing/ADRs/back-end/specificity-db-columns.md (hunk 1 above)
Comment (nunogois) [reply website/docs/contributing/ADRs/back-end/specificity-db-columns.md:12]: Addressed in https://github.com/Unleash/unleash/pull/5499/commits/c5f7fd38c4e8352084a6787e0e5ada127a8d4378 - Let me know if it helps 👍

File: website/docs/contributing/ADRs/back-end/specificity-db-columns.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+The adoption of this practice comes with minimal concerns:
+
+1. **Slight increase in verbosity**: The queries will be slightly longer due to the addition of table names in columns. However, the benefits in clarity and maintainability far outweigh this minor increase in verbosity.
+
+2. **Initial adaptation curve**: There might be a brief period of adjustment as developers adapt to this new standard. However, given its straightforward nature, this learning curve is expected to be minimal.
+
+## Conclusion
+
+The adoption of explicit table name specification in our SQL queries is a strategic decision aimed at improving the clarity, maintainability, and reliability of our database interactions. This practice will ensure that our queries remain robust and clear, especially in the context of evolving database schemas and complex query scenarios. We will gradually implement this standard in our existing codebase by following the Girl Scout Rule. This change aligns with our commitment to writing clean, understandable, and maintainable code, thereby enhancing the overall quality of our software development processes.
```
Comment (chriswk) [inline website/docs/contributing/ADRs/back-end/specificity-db-columns.md:75]: I assume Girl Scout Rule is a modernization of the Boy scout rule concept to be more gender positive

**Review by chriswk — APPROVED** [review]
Looks good. I'm all for clarity over brevity

File: website/docs/contributing/ADRs/back-end/specificity-db-columns.md (hunk 3 above)
Comment (kwasniew) [reply website/docs/contributing/ADRs/back-end/specificity-db-columns.md:75]: That was my suggestions. I think the language in general is very biased towards masculine forms and would like to shift the balance.

**Review by thomasheartman — APPROVED** [review]

**Pull request merged**